push/CMakeLists.txt

```diff
@@ -21,7 +18,6 @@ target_link_libraries(push
     ${PROJECT_NAME}::core
   PRIVATE
     ${CMAKE_THREAD_LIBS_INIT}
-    ${CURL_LIBRARIES}
```

Comment:
I moved all CURL references to the CPR package scripts, because the push library does not directly depend on it. Instead the CPR library has the dependency to CURL and should provide this as a transitive dependency.